c/busybox-1.22.0/date_false-unreach-call_true-no-overflow_true-valid-memsafety.c

```diff
@@ -1642,6 +1642,18 @@ static void * xmalloc(unsigned long int size)
   return ptr;
 }
 
+// model stat function, in this benchmark there is used
+// only the st_mtim.tv_sec variable
+int stat(const char *__file, struct stat *__buf)
+{
+  __buf->st_mtim.tv_sec = __VERIFIER_nondet_long();
+  if (__VERIFIER_nondet_char())
```

Comment:
Why use `nondet_char` instead of `nondet_int` here?
The if could also be moved to the beginning of the function.

Reply:
As it stands in the commit message: "The definition used is taken from the who benchmark", so that is the reason why there is `nondet_char`. Does it matter? `nondet_bool` would fit even better. Yes, the if could be at the beginning.